ciao-controller/instance.go

```diff
@@ -359,8 +359,8 @@ func newConfig(ctl *controller, wl *types.Workload, instanceID string, tenantID
 		userData.Hostname = "cnci-" + tenantID
 	}
 
-	// handle storage resources in workload definition
-	if wl.Storage != nil {
+	// if no instance volumes specified, use those in workload
+	if len(volumes) == 0 && wl.Storage != nil {
```

Comment:
Okay. So what should happen in this case where the user tries to override the boot device? 

Should we just reject this at instance creation time with a check (banning boot_index=0) ?

Or are you proposing we replace the boot device in the workload with the new one?